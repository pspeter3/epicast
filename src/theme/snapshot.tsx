import * as React from "react";
import TestRenderer from "react-test-renderer";

export const snapshot = (name: string, SFC: React.SFC<{}>): void => {
    describe(name, () => {
        it(`should match snapshot`, () => {
            const component = TestRenderer.create(<SFC>Snapshot</SFC>);
            const tree = component.toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
};
