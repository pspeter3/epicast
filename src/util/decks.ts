import { Deck } from "../core/types";
import { size } from "./stacks";

export const unique = (deck: Deck): string[] =>
    Object.keys(
        deck.reduce(
            (set, stack) => {
                return Object.keys(stack).reduce((collection, key) => {
                    collection[key] = null;
                    return collection;
                }, set);
            },
            {} as Record<string, null>,
        ),
    );

export const total = (deck: Deck): number => deck.reduce((sum, stack) => sum + size(stack), 0);
