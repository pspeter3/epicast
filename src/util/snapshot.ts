import { shallow } from "enzyme";
import * as React from "react";

export type Factory = () => React.ReactElement<any>;

export const snapshot = (factory: Factory): (() => void) => {
    return () => {
        expect(shallow(factory())).toMatchSnapshot();
    };
};

export const snapshotSuite = (name: string, factory: Factory): void => {
    describe(name, () => {
        it("should match snapshot", snapshot(factory));
    });
};
