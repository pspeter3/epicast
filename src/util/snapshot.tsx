import * as React from "react";
import TestRenderer from "react-test-renderer";

export type Factory = () => React.ReactElement<any>;

export const snapshot = (factory: Factory): (() => void) => {
    return () => {
        const component = TestRenderer.create(factory());
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    };
};

export const snapshotSuite = (name: string, factory: Factory): void => {
    describe(name, () => {
        it("should match snapshot", snapshot(factory));
    });
};
