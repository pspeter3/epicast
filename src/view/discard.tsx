import * as React from "react";
import { IconButton } from "../theme/buttons";
import { TrashIcon } from "../theme/icons";
import { Main, Row } from "../theme/layout";
import { Text } from "../theme/typography";

export const Discard: React.SFC<{}> = () => {
    return (
        <Main>
            <Row>
                <Text>Atlanta</Text>
                <IconButton>
                    <TrashIcon />
                </IconButton>
            </Row>
            <Row>
                <Text>San Francisco</Text>
                <IconButton>
                    <TrashIcon />
                </IconButton>
            </Row>
            <Row>
                <Text>San Francisco</Text>
                <IconButton>
                    <TrashIcon />
                </IconButton>
            </Row>
        </Main>
    );
};
