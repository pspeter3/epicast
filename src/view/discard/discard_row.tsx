import * as React from "react";
import { IconButton } from "../../theme/buttons";
import { TrashIcon } from "../../theme/icons";
import { Row } from "../../theme/layout";
import { Text } from "../../theme/typography";

export interface Props {
    city: string;
    onRemove: (city: string) => void;
}

export class DiscardRow extends React.PureComponent<Props, {}> {
    public render() {
        const { city } = this.props;
        return (
            <Row>
                <Text>{city}</Text>
                <IconButton onClick={this._onRemove}>
                    <TrashIcon />
                </IconButton>
            </Row>
        );
    }

    private _onRemove = () => {
        this.props.onRemove(this.props.city);
    };
}
