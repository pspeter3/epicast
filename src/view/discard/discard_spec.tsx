import * as React from "react";
import { snapshotSuite } from "../../util/snapshot";
import { Discard } from "./discard";

snapshotSuite("Discard", () => (
    <Discard discard={{ "San Francisco": 2, "Los Angeles": 1 }} onRemove={jest.fn()} />
));
