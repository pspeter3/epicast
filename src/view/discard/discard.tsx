import * as React from "react";
import { Stack } from "../../core/types";
import { Main } from "../../theme/layout";
import { DiscardRow } from "./discard_row";

export interface Props {
    discard: Stack;
    onRemove: (city: string) => void;
}

export class Discard extends React.PureComponent<Props, {}> {
    public render() {
        const { discard, onRemove } = this.props;
        return (
            <Main>
                {Object.keys(discard)
                    .sort()
                    .reduce(
                        (children, city) => {
                            const count = discard[city];
                            for (let i = 0; i < count; i++) {
                                children.push(
                                    <DiscardRow
                                        key={`${city}${i}`}
                                        city={city}
                                        onRemove={onRemove}
                                    />,
                                );
                            }
                            return children;
                        },
                        [] as Array<React.ReactElement<{}>>,
                    )}
            </Main>
        );
    }
}
