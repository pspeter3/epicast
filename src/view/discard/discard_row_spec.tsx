import { shallow } from "enzyme";
import * as React from "react";
import { IconButton } from "../../theme/buttons";
import { snapshot } from "../../util/snapshot";
import { DiscardRow } from "./discard_row";

describe("DiscardRow", () => {
    const createProps = () => {
        return {
            city: "San Francisco",
            onRemove: jest.fn(),
        };
    };

    it("should match snapshot", snapshot(() => <DiscardRow {...createProps()} />));

    it("should handle click", () => {
        const props = createProps();
        const tree = shallow(<DiscardRow {...props} />);
        tree.find(IconButton).simulate("click");
        expect(props.onRemove).toHaveBeenCalledWith(props.city);
    });
});
