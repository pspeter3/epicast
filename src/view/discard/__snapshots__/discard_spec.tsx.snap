// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Discard should match snapshot 1`] = `
<Component>
  <DiscardRow
    city="Los Angeles"
    key="Los Angeles0"
    onRemove={[MockFunction]}
  />
  <DiscardRow
    city="San Francisco"
    key="San Francisco0"
    onRemove={[MockFunction]}
  />
  <DiscardRow
    city="San Francisco"
    key="San Francisco1"
    onRemove={[MockFunction]}
  />
</Component>
`;
